draft-kazuho-httpbis-priority.md

```diff
@@ -233,18 +233,78 @@ set to `3` and the progressive parameter set to `1`.
+  therefore only a suggestion.
+
+In practice, this approach is quite one-sided and puts a burden on the endpoint
+with the most limited amount of information. A client processing HTML and
+generating requests generally has only a URL and a type (determined by an HTML
+element or other mechanism) on which to make a prioritization prediction.
+
+A server typically has more information about the resource than the client,
+including the context around it (such as linked representations). It uses local
+knowledge, with input from the client, to generate a single selected
+representation ({{?RFC7231}}, Section 3) and respond. During response
+generation, the server can possibly determine that the client's predicted
+prioritization was incorrect (perhaps sub-optimal) for how the selected
+representation will be used. An HTTP/2 server might use this information,
+together with other inputs, to send the response at a different effective
+priority than the client suggested. However, it has no explicit means on which
+to present such a finding to the client.
+
+The frame-based prioritization process weakens further when intermediaries are
+considered because only the client is permitted to provide an explicit signal.
+
+## A Collaborative Model
+
+A collaborative model is presented that attempts to abstract priority and
+collaboration from concerns about HTTP version and active connections.
+
+Resources have an initial priority that begins with a neutral value. Requests
+and responses permit collaborative changes to the resources initial priority
+information. A client attaches suggested priority information to the request,
+omission activates a default priority. A server can (if supported) attach
+priority information to the response, omission might be an acceptance of the
+client offer or or implicit ignore. Once a request or response is in flight, an
+endpoint cannot modify the initial priority, further changes are classed as
+reprioritization that requires a non-request or non-response carriage mechanism
+(e.g. a frame).
+
+As explained above, in this model HTTP/2 does not provide an explicit
+prioritization signal for servers.
+
+## Explicit Client and Server Collaboration {#collaboration}
+
+The "Priority" header field can be attached to responses in order to provide an
+explicit signal that allows servers collaborate to collaborate on the
+prioritization process. A server sends new priority parameter values or confirms
+the client suggestion. This process works well in the case of intermediaries
+because the client suggested priorities are passed to the origin, giving it the
+opportunity to improve the priority information before the intermediary
+allocates resources to the response.
+
+For example, a client using an HTML document discovers several inline images and
+requests them with a low urgency via an intermediary. The origin server has
+knowledge that one of the images heavily affects the usage of the HTML document
```

Comment:
> heavily affects the usage of the HTML document

heavily impacts the end user's experience

Reply:
I stuck with use of HTML in order to align with @kazuho terminology.

I think we have an overarching appreciation of user experience but focus on individual resource usage in specific examples.

Reply:
While I am all in favor of consistency, I don't think the sentence as it stands makes much sense semantically... Maybe changing to "the way the HTML document is perceived by the end user" could help? or "the practical usage of the HTML document"?